libs/types-common/src/Data/Nonce.hs

```diff
@@ -70,3 +77,24 @@ instance Cql Nonce where
   ctype = Tagged UuidColumn
   toCql = toCql . unNonce
   fromCql v = Nonce <$> fromCql v
+
+newtype NonceTtlSecs = NonceTtlSecs {unNonceTtlSecs :: Word32}
+  deriving (Eq, Show, Generic)
+  deriving (A.FromJSON, A.ToJSON, S.ToSchema) via (Schema NonceTtlSecs)
+
+-- | Convert 'Word32' to 'Int32', with clipping if it doesn't fit.
+word32ToInt32 :: Word32 -> Int32
+word32ToInt32 = fromIntegral . min (fromIntegral (maxBound @Int32))
```

Comment:
i know it's an impractical corner case in *this* context, but hum...  can we use `Range` to make sure we don't run into this overflow?